cdap-master-spi/src/main/java/io/cdap/cdap/master/spi/environment/MasterEnvironment.java

```diff
@@ -112,4 +112,11 @@ default void onNamespaceCreation(String namespace, Map<String, String> propertie
   default void onNamespaceDeletion(String namespace, Map<String, String> properties) throws Exception {
     // no-op by default
   }
+
+  /**
+   * Syncs configs across namespaces. Called when appfabric starts up.
+   */
+  default void syncNamespaceConfigs(Map<String, Map<String, String>> namespaceConfigs) throws Exception {
```

Comment:
It seems like synchronizing the CConf between namespaces is an implementation detail of Kubernetes.

If it's not easy to put all of the logic in KubeMasterEnvironment (since it's used everywhere), maybe it would be good to generalize this to an app-fabric initialization method instead, i.e. initializeMaster method which is only called in AppFabric?